### ISSUE TITLE
Error: This is R 3.2.2, package 'S4Vectors' needs >= 3.3.0

### ISSUE THREAD

**Issue by JoKw** [bcbio/bcbio-nextgen#1898]
Hi,

When I ran bubbletree, it gave me the error below:

Loading required package: S4Vectors
Error: This is R 3.2.2, package 'S4Vectors' needs >= 3.3.0
In addition: Warning messages:
1: package 'IRanges' was built under R version 3.3.1
2: package 'BiocGenerics' was built under R version 3.3.1

Any solution? Thanks

**Comment by chapmanb** [comment]
Thanks for the report, I believe you are getting R packages compiled with a different version of R into your bcbio run. bcbio installs R and the dependencies and is currently based around 3.2. I pushed a fix to the latest development version which I hope will avoid this by ignoring `R_LIBS`. You can update with:
```
bcbio_nextgen.py upgrade -u development
```
and re-run in place. Hope this fixes it for you.

**Comment by JoKw** [comment]
Sorry, Brad, I still got the same error after updating. I think 'S4Vectors' needs >= 3.3.0 while the R installed is only 3.2

**Comment by chapmanb** [comment]
Thanks much for testing this rorry about the continued problems. It looks like we're running into issues where we load both R 3.2 and 3.3 libraries through conda. Could you try doing:
```
bcbio_conda install -c bioconda 'bioconductor-biocgenerics=0.18.0' 'bioconductor-iranges=2.4.7'
```
to get R 3.2 compatible versions of the problem libraries? I hope this will resolve this for you in the short term.

In the slightly longer term, R 3.4 and a new bioconductor release are due out in a week and we'll work to move all the R dependencies in bcbio over to use this updated version. I'll leave this open to discuss the work on transitioning over to that and correctly fixing the incompatibilities. Thanks again for the patience and help debugging.

**Comment by JoKw** [comment]
Thanks, Brad. I finally find the cause of the problem. In the prep-run.R under the bubbletree folder created for each sample, the first line is 

.libPaths(c("/usr/local/lib/R/site-library")). 

If I modify it to 

.libPaths(c("/usr/local/share/bcbio/anaconda/lib/R/library"), 

then the error disappears. But sadly, I encounter another error: 

Error in limma::weighted.median(lrr, seg.size, na.rm = TRUE) :
  NA weights not allowed
Calls: btpredict ... with -> with.default -> eval -> eval -> <Anonymous>
Execution halted

Is it related to my tumor-only sample?

**Comment by chapmanb** [comment]
Thanks for identifying the import problem, that is a holdover from when we installed tools via custom R installs. Since we're now using conda for installs, I swapped it over in latest development to ensure we load only the anaconda library. Hopefully that will avoid the original problem going forward.

Your new error is a little bit trickier and sounds like BubbleTree is failing from having too little inputs at some point since it's trying to calculate a median on an empty input. We've tried to make BubbleTree relatively robust to these cases but have definitely not caught them all. My guess here is that your tumor-only sample might have an issue with either the CNV or variant call inputs. Are you using CNVkit and VarDict for the callers or other approaches? Do the outputs look reasonable?

We're working actively now to improve tumor-only difficult cases, with the discussion taking place in #1876 so hope this will be better in the future. Sorry to not have an exact solution but hopefully this helps.
